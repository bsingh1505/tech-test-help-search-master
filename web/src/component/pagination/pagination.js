import { Button } from "@mui/material";
import { ITEMS_PER_PAGE } from "../../utils/constants";

export const PaginationView = ({ totalItems, currentPage = 1, setCurrentPage }) => {
 

  if (!totalItems || totalItems === 0) return null;
  const roundUp = (totalItems % ITEMS_PER_PAGE) === 0 ? 0 : 1;
  const paginationCount = Math.floor(totalItems / ITEMS_PER_PAGE) + roundUp;
  const paginationArray = [...Array(paginationCount).keys()];

  const handleClick = (selectedPage) => {
    setCurrentPage(selectedPage);
  };

  return (
    <div className="pagination">
      {paginationArray.map((item) => (
        <Button
          key={`btn-${item + 1}`}
          data-testid={`btn-${item + 1}`}
          variant={item === currentPage ? "contained" : "outlined"}
          color="primary"
          onClick={() => handleClick(item)}
        >
          {item + 1}
        </Button>
      ))}
    </div>
  );
};
