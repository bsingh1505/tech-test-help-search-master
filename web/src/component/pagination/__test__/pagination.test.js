import TestRenderer from "react-test-renderer";
import { render, fireEvent, screen } from "@testing-library/react";
import { PaginationView } from "../pagination";

describe("Card component", () => {
  test("match snapshot : if totalItems is null", () => {
    const tree = TestRenderer.create(<PaginationView />);
    expect(tree).toMatchSnapshot();
  });
  test("match snapshot : For valid data", () => {
    const setCurrentPage = jest.fn();
    const tree = TestRenderer.create(<PaginationView totalItems={5} currentPage={1} setCurrentPage={setCurrentPage} />);
    expect(tree).toMatchSnapshot();
  });
  test("Pagination button being clicked and correct function being called", () => {
    const setCurrentPage = jest.fn();
    render(<PaginationView totalItems={5} currentPage={1} setCurrentPage={setCurrentPage} />);
    const btn = screen.getByTestId("btn-1");
    fireEvent.click(btn);
    expect(setCurrentPage).toHaveBeenCalledTimes(1);
  });
});
