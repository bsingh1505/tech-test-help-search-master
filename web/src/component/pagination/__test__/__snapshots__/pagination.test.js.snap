// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component match snapshot : For valid data 1`] = `
<div
  className="pagination"
>
  <button
    className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
    data-testid="btn-1"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    1
  </button>
</div>
`;

exports[`Card component match snapshot : if totalItems is null 1`] = `null`;
